js/controllers/books.js

```diff
@@ -0,0 +1,112 @@
+/* global app */
+/* global angular */
+/*global $rootScope*/
+
+app.controller('booksCtrl', ['$scope', '$state', '$stateParams', 'bookService', 'authorService', 'enableEditService', 'uiGridConstants', function($scope, $state, $stateParams, bookService, authorService, enableEditService, uiGridConstants, $index) {
+  
+  $scope.include = function(obj) {
+     bookService.updateBooks(obj);
+   };
+   
+  $scope.authorNameDropdown = authorService.authorsList; 
+  $scope.bookTemp = bookService.booksList;
+
+  $scope.genreTypes = [
+    {id:"Murder Mystery", value:"Murder Mystery"},
+    {id:"Historical Fiction", value:"Historical Fiction"},
+    {id:"Contemporary Fiction", value:"Contemporary Fiction"} 
+  ];
+  
+  $scope.genreTypesModal = [
+    "Murder Mystery", "Historical Fiction", "Contemporary Fiction" 
+  ];
+  
+  $scope.edit = enableEditService.edit;
+  
+  $scope.bookGridOptions = {
+    
+    data: 'bookTemp',
+    enableCellSelection: true,
+    enableCellEdit: true,
+    enableCellEditOnFocus: true,
+    rowEditWaitInterval: -1,
+    saveFocus: false,
+    saveScroll: true,
+    saveGroupingExpandedStates: true,
+    enableRowSelection: true,
+    onRegisterApi: function (gridApi) {
+                   $scope.gridApi = gridApi;
+                   $scope.gridApi.edit.on.afterCellEdit($scope,function(rowEntity, colDef, newValue, oldValue){
+                         if(newValue != oldValue){
+                            $state.reload();
+                         }
+                    });
+                   },
+  
+    columnDefs: [
+
+       { field: 'id', displayName: 'ID', enableCellEdit: false, enableHiding: false
+       },
+       { field: 'name', displayName: 'Name', enableHiding: false, cellEditableCondition: $scope.edit
+       },
+       { field: 'author', displayName: 'Author', enableHiding: false, cellEditableCondition: $scope.edit,
+         cellTemplate: 'tempAuthor.html', 
+         editableCellTemplate: 'ui-grid/dropdownEditor',
+         editDropdownOptionsArray: authorService.authorsList,
+         editDropdownIdLabel: 'id',
+         editDropdownValueLabel: 'name' 
+       },
+       { field: 'genre', name: 'genre', displayName: 'Genre', enableHiding: false, cellEditableCondition: $scope.edit,
+         editableCellTemplate: 'ui-grid/dropdownEditor',
+         editDropdownOptionsArray: $scope.genreTypes
+       },
+       { field: 'dateAdded', displayName: 'Date Added', enableCellEdit: false, enableHiding: false
+       },
+       { field: 'outOfPrint', displayName: 'Out of Print?', enableCellEdit: false, enableHiding: false
+       },
+    ]
+  };
+  
+$scope.bookAuthorNames = $scope.bookTemp.map(function(book){
+     var authorNum = book.author; 
+     
+     $scope.authorName = authorService.authorsList.filter(function(author){
+        return author.id == authorNum; 
+    })[0].name;
+    
+    return $scope.authorName;
+ });
+
+$scope.authorNameArray = authorService.authorsList.map(function(author){
```

Comment:
It doesn't look like this function is used anwhere - please check. If it is not used, please remove or provide an explanatory comment about the cases where it could be useful in the future